examples/ring_topology.rs

```diff
@@ -0,0 +1,42 @@
+use rr_channel::detthread;
+use rr_channel::crossbeam;
+use std::sync::Mutex;
+use std::sync::Arc;
+
+fn main() {
+    let mut senders: Vec<Mutex<_>> = Vec::new();
+    let mut receivers: Vec<Mutex<_>> = Vec::new();
+
+    for _ in 0..10 {
+        let (s, r) = crossbeam::unbounded::<i32>();
+        senders.push(Mutex::new(s));
+        receivers.push(Mutex::new(r));
+    }
+    
+    let senders = Arc::new(senders);
+    let receivers = Arc::new(receivers);
+
+    for i in 0..10 {
+        // Clone via arc for moving into closure.
+        let senders = senders.clone();
+        let receivers = receivers.clone();
+        rr_channel::detthread::spawn(move || {
+            //Set up the ring
+            if i == 0 {
+                let sender = senders[0].try_lock().expect("Lock already taken");
+                sender.send(1);
+            }
+
+            // Locks should never block as every thread accesses exclusive members of vector.
+            let sender = senders[(i + 1) % 10].try_lock().expect("Lock already taken");
+            let receiver = receivers[(i) % 10].try_lock().expect("Lock already taken");
+            // Send message around ring.
+            let result = receiver.try_recv().unwrap();
```

Comment:
`try_recv()` does not block waiting for the message to arrive, which is what we want. Currently you're seeing a panic because `try_recv()` will return an error and then `.unwrap()` causes a pain.

In general I would use `.expect("Appropriate error")` instead of `.unwrap()`.

Idk how familiar you're with Rust's error handling facilities: `.unwarp()`, `panic`, `?`, `Result`, etc. If not we can talk about some of them together.